core/corehttp/gateway_handler.go

```diff
@@ -292,6 +294,10 @@ func (i *gatewayHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	defer cancel()
 	r = r.WithContext(ctx)
 
+	// Emit tracing context headers if present
+	prop := otel.GetTextMapPropagator()
+	prop.Inject(r.Context(), propagation.HeaderCarrier(w.Header()))
+
```

Comment:
Do you know the difference with:
```go
		gw = otelhttp.NewHandler(gw, "Gateway.Request", otelhttp.WithPropagators(tracing.Propagator()))
```
[There ?](https://github.com/ipfs/kubo/blob/d1713ca28e939fcc611e093b5bfcb1e3a54c5b14/core/corehttp/gateway.go#L55)

We already use this for cmds and this works great: https://github.com/ipfs/kubo/blob/d1713ca28e939fcc611e093b5bfcb1e3a54c5b14/core/corehttp/commands.go#L150

They seems equivalent to me, except `otelhttp` is cleaner and more descriptive.